conda_smithy/linter/utils.py

```diff
@@ -185,3 +206,15 @@ def _lint_package_version(version: Optional[str]) -> Optional[str]:
         VersionOrder(ver)
     except InvalidVersionSpec as e:
         return invalid_version.format(ver=ver, err=e)
+
+
+@lru_cache(maxsize=1)
```

Comment:
What's the lifetime of this cache? I'm not too sure how the linter is currently set up, but if it's a long-running deployment, that would basically mean we have to redeploy to pick up changes in the `hints.toml` from the pinning repo...?

Reply:
Good point. We should use something like this (https://stackoverflow.com/a/63674816), though I think that specific code has edge cases when it could avoid them.